### DISCUSSION TITLE
PGroongaで Index only scan を使うにはどうしたらいいか？

### DISCUSSION THREAD

**Discussion by komainu8** [groonga/groonga#1584] [グルカイ！]
PGroongaはインデックスの内部にPostgreSQLに格納されているインデックスの元の文書を保持していると思います。
それらは、`Index only scan`を実現するために保持していると思っていますが、PGroongaのインデックスを設定したカラムに対する検索で`Index only scan`を使う条件みたいなものはあるのでしょうか？
(通常は`Index scan`を使っているように見えます。)

**Comment by askdkc** [comment]
本日の動画🎥

https://www.youtube.com/live/JkvCRPDxBeg?feature=share

今日の内容の参考文献📙はこの辺かしら？

https://www.postgresql.org/docs/current/indexes-index-only-scans.html

> 1. The index type must support index-only scans.

> 2. The query must reference only columns stored in the index.

> If these two fundamental requirements are met, then all the data values required by the query are available from the index, so an index-only scan is physically possible.

**Reply by kou** [reply]
その後の↓も大事なんです。で、今日は↓の話をしていました。

> But there is an additional requirement for any table scan in PostgreSQL: it must verify that each retrieved row be “visible” to the query's MVCC snapshot, as discussed in [Chapter 13](https://www.postgresql.org/docs/current/mvcc.html).

**Reply by askdkc** [reply]
@kou さん

うぉーん。

情報不足で申し訳ない🥲
